contentprovider.go

```diff
@@ -306,8 +329,8 @@ func matchScore(secs []DocumentSection, m *LineMatch) float64 {
 			score = scorePartialWordMatch
 		}
 
-		sec := findSection(secs, f.Offset, uint32(f.MatchLength))
-		if sec != nil {
+		if f.symbolSectionIdx != nil {
+			sec := secs[*f.symbolSectionIdx]
```

Comment:
It looks like previously we called findSection. But you moved it to to happening somewhere else. I'm trying to understand why that is preferred? Is it to avoid calling findSymbol more than once?

Maybe I misunderstood the implications in an earlier version of this code. But it seems like the dumb thing of just calling findSymbol here would do the same amount of work but avoid the indirection?

Sorry I have a meeting now and don't know if I'll get back to this PR before next week. So I am gonna pose this half baked comment now rather than confirming my thoughts.

Reply:
That's correct. I wanted to avoid calling findSection twice, because it is effecively a binary search. We could revert to the original code if we don't mind the additional call.

Reply:
Note: we chatted about this over zoom. Decided to make matchScore be part of content provider to avoid state.